program/rust/src/rust_oracle.rs

```diff
@@ -169,6 +171,69 @@ pub fn add_price(
     Ok(SUCCESS)
 }
 
+pub fn init_price(
+    program_id: &Pubkey,
+    accounts: &[AccountInfo],
+    instruction_data: &[u8],
+) -> OracleResult {
+    let cmd_args = load::<cmd_init_price_t>(instruction_data)?;
+
```

Comment:
The check that `instruction_data.len() == size_of::<cmd_init_price_t>(cmd_init_price)` went away because `load` will work as long as `instruction_data.len() >= size_of::<cmd_init_price_t>(cmd_init_price)`

Reply:
Just flagging the differences with C, I think this is acceptable.